test/distributed/test_device_mesh.py

```diff
@@ -270,6 +270,29 @@ def test_set_mesh_dim_group_options(self):
         # Fake pg only have BackendType as BackendType::CUSTOM.
         self.assertEqual(mesh.get_group(1)._get_backend_name(), "custom")
 
+    @with_comms(eager_init=True)
```

Comment:
we can use `@with_comms` unless you do need to eagerly init, since the fix (https://github.com/pytorch/pytorch/pull/139637) has been merged.

Reply:
i was getting a lot of nccl warnings about device_id not being specified that went away when i added eager_init=True